docs/site/content/en/docs/Tasks/define-functions/index.md

```diff
@@ -0,0 +1,95 @@
+date: 2024-02-29
+weight: 4
+---
+> **Prerequisites**: You have already
+
+> 1. [created a Test](/docs/tasks/create-new-test/)
+
+> 2. [uploaded](/docs/tasks/upload-run/) some data
+
+ Using `Functions` in `Horreum` is a feature that provides a great deal of bespoke functionality to `Horreum` that is under the control of a user. The ability to use a `Function` written in Javascript makes it possible in the web User Interface (UI) to have:
+
+These Functions can be categorized as:
+
+* Selector (filter)
+* Transforming
+* Combination
+
+ When using `Horreum` you will find in `Functions` used in these Horreum objects:
+ 
+* Label (Schema) Combination Function
+* Transformer (Schema) Combination Function
+* Experiment (Test) Filtering Selector
+* Compare (Test) URL Function
+* Change Detection (Test) Timeline filter Function
+* Change Detection (Test) Fingerprint filter Function
+* Change Detection (Test) Variable Calculation Function
+* View (Test) Component Rendering Function
+* Filtering (Report) Function
+* Category (Report) Function
+* Category formatter (Report) Function
+* Series (Report) Function
+* Series formatter (Report) Function
+* Scale Label (Report) Function
+* Scale Label formatter (Report) Function
+* Component Transformation (Report) Function
+
+## Making use of Horreum Functions
+
+ Javascript [ECMAScript 2023](https://262.ecma-international.org/14.0/) specification is available throughout Horreum `Functions`. A reference for Javascript functions based use is available [here](https://www.w3schools.com/jsref/default.asp)
+
+## Example Filtering Functions
+
+These Functions rely on a condition evaluation to return a boolean value.
+The following will filter based on the individual Label Extractor only having the value 72.
+
+```javascript
+value => value === 73
+```
+
+## Example Transforming Functions
+
+`Transforming Functions` rely on a returned value that is an Object, Array or individual value.
+This `Function` relies on 12 `Extractors` setup on the `Schema Label`. Each Extractor configured to obtain an array of data items (except buildId and buildUrl).
+
+```javascript
+({runtimeName, rssStartup, maxRss, avBuildTime, avTimeToFirstRequest, avThroughput, rssFirstRequest, maxThroughputDensity, buildId, buildUrl, quarkusVersion, springVersion}) => {
+    var map = runtimeName.map((name, i) => ({
+        runtime: name.split('-')[0],
+        buildType: name.split('-')[1],
+        rssStartup: rssStartup[i],
+        maxRss: maxRss[i],
+        avBuildTime: avBuildTime[i],
+        avTimeToFirstRequest: avTimeToFirstRequest[i],
+        avThroughput: avThroughput[i],
+        rssFirstRequest: rssFirstRequest[i],
+        maxThroughputDensity: maxThroughputDensity[i],
+        buildId: buildId,
+        buildUrl: buildUrl,
+        version: ((name.split('-')[0].substring(0, 6) == 'spring' ) ? springVersion: quarkusVersion )
+    }))
+    return map;
+}
+``` 
+
```

Comment:
Can we have abridged json documents showing input & output from this function?

Reply:
sure